src/components/weather/Weather.jsx

```diff
@@ -0,0 +1,93 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+import {
+  InputCitySearch,
+  TempWrapper,
+  WeatherWrapper,
+} from "../style/Elements";
+import { SecBtn, TransparentBtn } from "../style/Button";
+
+const Weather = () => {
+  const [coordinates, setCoordinates] = useState({ latitude: 0, longitude: 0 });
+  const [weather, setWeather] = useState(null);
+  const [city, setCity] = useState(localStorage.getItem("city"));
+  const [isInputVisible, setIsInputVisible] = useState(false);
+
+  const getWeatherData = async () => {
+    try {
+      const response =
+        coordinates.latitude !== 0 &&
+        (await axios.get(
+          `https://api.openweathermap.org/data/2.5/weather?${
+            city !== null
+              ? `q=${city}`
+              : `lat=${coordinates.latitude}&lon=${coordinates.longitude}`
+          }&appid=021cbeb6524e6c10b8d5ac497d7e73a3`
```

Comment:
Can store this private data in an env variable otherwise the weather component might not work properly due to misuse of secret API key.